source/engine/hdf5/HDF5WriterP.cpp

```diff
@@ -0,0 +1,316 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * HDF5WriterP.cpp
+ *
+ *  Created on: March 20, 2017
+ *      Author: Junmin
+ */
+
+
+#include <iostream> //needs to go away, this is just for demo purposes
+
+#ifdef ADIOS_HAVE_MPI
```

Comment:
No need for this #ifdef here.  This file won't be part of the build if it's not supported.